src/main/java/edu/cornell/kfs/fp/document/web/struts/RecurringDisbursementVoucherAction.java

```diff
@@ -61,6 +68,44 @@ public ActionForward copy(ActionMapping mapping, ActionForm form, HttpServletReq
     	recurringForm.getPdpStatuses().clear();
     	ActionForward forward = super.copy(mapping, form, request, response);
         return forward;
+        
+    }
+    
+    public ActionForward confirmAndCancel(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) throws Exception {
+        RecurringDisbursementVoucherForm recurringForm = (RecurringDisbursementVoucherForm) form;
+        RecurringDisbursementVoucherDocument recurringDV = (RecurringDisbursementVoucherDocument) recurringForm.getDocument();
+        
+        if (StringUtils.isEmpty(recurringDV.getPaymentCancelReason())) {
+            String paymentCancelFieldName = AccountingDocumentRuleBaseConstants.ERROR_PATH.DOCUMENT_ERROR_PREFIX + 
+                    CUKFSPropertyConstants.RECURRING_DV_PAYMENT_CANCEL_REASON_FIELD_NAME;
+            GlobalVariables.getMessageMap().putError(paymentCancelFieldName, CUKFSKeyConstants.ERROR_RCDV_PAYMENT_CANCEL_REASON_REQUIRED);
+        } else {
+            Set<String> canceledPaymentGroups = getRecurringDisbursementVoucherDocumentService().cancelPDPPayments(recurringDV, recurringDV.getPaymentCancelReason());
+            Set<String> canceledDVs = getRecurringDisbursementVoucherDocumentService().cancelDisbursementVouchers(recurringDV, recurringDV.getPaymentCancelReason());
+            reportCanceledItemsToBrowser(canceledPaymentGroups, canceledDVs);
+        }
+        forceRefreshOfPDPStatuses(recurringForm);
```

Comment:
Is it possible to get here even though some error occurred during cancelation? then the pdp statuses would be incorrectly updated to null